README.md

```diff
@@ -192,7 +192,7 @@ A Service Binding describes the connection between a [Provisioned Service](#prov
 
 Restricting service binding to resources within the same namespace is strongly **RECOMMENDED**.  Cross-namespace service binding **SHOULD** be secured appropriately by the implementor to prevent attacks like privilege escalation and secret enumeration.
 
-A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference`-like declaration to a `PodSpec`-able resource.  A `ServiceBinding` **MAY** define the application reference by-name or by-[label selector][ls]. A name and selector **MUST NOT** be defined in the same reference.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference`-like declaration to a Provisioned Service-able resource.
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference`-like declaration to a `PodSpec`-able resource.  A `ServiceBinding` **MAY** define the application reference by-name or by-[label selector][ls]. A name and selector **MUST NOT** be defined in the same reference.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference`-like declaration to a Provisioned Service-able resource.  Extensions and implementations **MAY** allow additional kinds of applications and services to be referenced.
```

Comment:
Why include "services" here? I don't see any further details about referencing "services" in other ways under the custom projections section.

Also the word "kinds of" reads as this is something about Kubernetes resource kinds. Probably the sentence can be rephrased something like this:
`Extensions and implementations **MAY** allow additional mechanisms for referencing applications`
or
`Extensions and implementations **MAY** allow additional ways for referencing applications`

Reply:
> Why include "services" here? I don't see any further details about referencing "services" in other ways under the custom projections section.

See https://github.com/k8s-service-bindings/spec#binding-secret-generation-strategies

> Also the word "kinds of" reads as this is something about Kubernetes resource kinds.

Correct. Other Kubernetes resource types (kinds) may be referenced. We are **not** encuraging other ways to define the reference.

Reply:
Thanks for the clarification!

Reply:
@baijum Are you good with the following or you think it needs to be more clear?
> Extensions and implementations **MAY** allow additional kinds of applications and services to be referenced.

Reply:
I am good with these changes!